snooty/postprocess.py

```diff
@@ -470,15 +470,23 @@ class HeadingData(NamedTuple):
         depth: int
         id: str
         title: Sequence[n.InlineNode]
-        selector_id: Optional[str]
+        selector_ids: Dict[Any, Any]
 
     def __init__(self, context: Context) -> None:
         super().__init__(context)
         self.contents_depth = sys.maxsize
         self.current_depth = 0
         self.has_contents_directive = False
         self.headings: List[ContentsHandler.HeadingData] = []
-        self.scanned_pattern: List[str] = []
+        self.scanned_pattern: List[Tuple[str, str]] = []
+
+    def scan_pattern(self, arr: List[Tuple[str, str]]) -> Dict[Any, Any]:
+        if arr is []:
```

Comment:
minor - can use `if not arr`. also wondering if preference is to have empty object or None

Reply:
Note: `is` is the identity operator, so this condition will *always* be false:

```
>>> [] is []
False
```